python/pyspark/pandas/tests/test_dataframe.py

```diff
@@ -1449,7 +1454,12 @@ def test_fillna(self):
         pdf.fillna({"x": -1, "y": -2, "z": -5}, inplace=True)
         psdf.fillna({"x": -1, "y": -2, "z": -5}, inplace=True)
         self.assert_eq(psdf, pdf)
-        self.assert_eq(psser, pser)
+        # SPARK-38946: Since Spark 3.4, fillna with inplace generate a new dataframe to follow
+        # pandas 1.4 behaviors
+        if LooseVersion(pd.__version__) >= LooseVersion("1.4"):
+            self.assert_eq(psser, pser)
```

Comment:
ditto.

Reply:
See https://github.com/apache/spark/pull/36353#issuecomment-1178956945 , change the version to 1.4.0-1.4.2